pkg/util/tracing/tracer.go

```diff
@@ -1127,10 +1120,10 @@ func (t *Tracer) startSpanFast(
 			recordingType = opts.minRecordingTypeOpt
 		}
 
-		shouldBeNoopSpan := !(t.AlwaysTrace() || opts.ForceRealSpan || recordingType != tracingpb.RecordingOff)
+		shouldBeNilSpan := !(t.AlwaysTrace() || opts.ForceRealSpan || recordingType != tracingpb.RecordingOff)
 		forceVerbose := t.forceOpNameVerbose(opName)
-		if shouldBeNoopSpan && !forceVerbose && !opts.Sterile {
-			return maybeWrapCtx(ctx, t.noopSpan)
+		if shouldBeNilSpan && !forceVerbose && !opts.Sterile {
+			return maybeWrapCtx(ctx, nil)
```

Comment:
@dhartunian  I ended up keeping these instead of just returning `ctx, nil`. It lead to some failied tests without. I'm guessing there are cases where ctx has a non nil span and the caller of startspan is expecting to clear that (or get its own span).